src/util/strencodings.cpp

```diff
@@ -444,13 +444,15 @@ bool ParseFixedPoint(std::string_view val, int decimals, int64_t *amount_out)
 std::string ToLower(std::string_view str)
 {
     std::string r;
+    r.reserve(str.size());
     for (auto ch : str) r += ToLower(ch);
```

Comment:
How about switching these additions to `[push_back](https://en.cppreference.com/w/cpp/string/basic_string/push_back)`?  [`+=`](https://en.cppreference.com/w/cpp/string/basic_string/operator%2B%3D) has no complexity guarantees, while push_back is guaranteed constant time.